### PR TITLE
DOC: exclude autogenerated c/cpp/html files from 'trailing whitespace' checks

### LINKED ISSUE #24526: DOC: code_checks patterns does not ignore .c files
should ignore .c files that are generated from .pyx (which are most)

get a bunch of errors like this

```
....
pandas/_libs/tslib.c:32200:Tailing whitespaces found:  * 
pandas/_libs/tslib.c:32201:Tailing whitespaces found:  * 
pandas/_libs/tslib.c:32211:Tailing whitespaces found:  * 
pandas/_libs/tslib.c:32213:Tailing whitespaces found:  * 
pandas/_libs/tslib.c:32214:Tailing whitespaces found:  * 
pandas/_libs/tslib.c:33594:Tailing whitespaces found:         
pandas/_libs/tslib.c:33595:Tailing whitespaces found:         
pandas/_libs/tslib.c:34701:Tailing whitespaces found:         
pandas/_libs/tslib.c:34702:Tailing whitespaces found:         
pandas/_libs/algos_take_helper.pxi:289:Tailing whitespaces found:     
pandas/_libs/algos_take_helper.pxi:324:Tailing whitespaces found:     
pandas/_libs/algos_take_helper.pxi:549:Tailing whitespaces found:     
pandas/_libs/algos_take_helper.pxi:584:Tailing whitespaces found:     
pandas/_libs/algos_take_helper.pxi:4709:Tailing whitespaces found:     
pandas/_libs/algos_take_helper.pxi:4744:Tailing whitespaces found:     
pandas.egg-info/PKG-INFO:10:Tailing whitespaces found: Description
pandas.egg-info/PKG-INFO:19:Tailing whitespaces found:         
pandas.egg-info/PKG-INFO:21:Tailing whitespaces found:         
pandas.egg-info/PKG-INFO:29:Tailing whitespaces found:         
pandas.egg-info/PKG-INFO:37:Tailing whitespaces found:         
pandas.egg-info/PKG-INFO:39:Tailing whitespaces found:         
pandas.egg-info/PKG-INFO:65:Tailing whitespaces found:         
pandas.egg-info/PKG-INFO:72:Tailing whitespaces found:         
Check that no file in the repo contains tailing whitespaces DONE
```

### PR DESCRIPTION
**Pull request by saurav2608**
- [x] closes #24526 
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
@saurav2608 no, just leave this for @datapythonista to look. The CI failure is unrelated.

**Comment by datapythonista** [comment]
There are C and C++ files in the pandas codebase, and I'd like to avoid tailing spaces on them too. Only the generated ones could be excluded.
```
./pandas/io/msgpack/_unpacker.cpp
./pandas/io/msgpack/_packer.cpp
./pandas/_libs/window.cpp
./pandas/util/move.c
./pandas/_libs/src/parser/io.c
./pandas/_libs/src/parser/tokenizer.c
./pandas/_libs/src/ujson/python/ujson.c
./pandas/_libs/src/ujson/python/objToJSON.c
./pandas/_libs/src/ujson/python/JSONtoObj.c
./pandas/_libs/src/ujson/lib/ultrajsondec.c
./pandas/_libs/src/ujson/lib/ultrajsonenc.c
./pandas/_libs/tslibs/src/datetime/np_datetime.c
./pandas/_libs/tslibs/src/datetime/np_datetime_strings.c
```

**Comment by WillAyd** [comment]
@saurav2608 can you address comments and merge master?

**Review by datapythonista — APPROVED** [review]
This makes sense to me, not ideal to exclude all the C/C++ from the tailing spaces check (may confuse a human reader), but I think we should validate everything we want this way, and it's probably the simplest approach.

I'd just do couple of things before merging:
- Adding a comment in the tailing spaces check (where the `--exclude` is) explaining that C/C++ files will be checked with the linting
- Add in you local environment (i.e. don't commit) tailing whitespaces to some file in `/pandas/io/msgpack` and see if the script (so the CI) fails in that case. I think so, but just to double check

Thanks for the fixes @saurav2608

**Comment by saurav2608** [comment]
> * Add in you local environment (i.e. don't commit) tailing whitespaces to some file in `/pandas/io/msgpack` and see if the script (so the CI) fails in that case. I think so, but just to double check
> 
@datapythonista 
The cpp files are autogenerated - `./pandas/io/msgpack/_unpacker.cpp` Can you confirm that? The linter checks only .c and .h files. I can add cpp too if cpp files here are not autogenerated.

**Comment by datapythonista** [comment]
You're right. I fail to see what we are doing here then. This PR is to exclude the autogenerated .cpp files from the tailing whitespace check, and we're then fully linting them?

**Comment by saurav2608** [comment]
> You're right. I fail to see what we are doing here then. This PR is to exclude the autogenerated .cpp files from the tailing whitespace check, and we're then fully linting them?

I have pushed the changes. The .c and .h files are fully Linted. I check and trailing whitespace does show up as an error.

**Review by jreback — CHANGES_REQUESTED** [review]

File: ci/code_checks.sh [hunk 1]
```diff
@@ -93,7 +93,7 @@ if [[ -z "$CHECK" || "$CHECK" == "lint" ]]; then
     # this particular codebase (e.g. src/headers, src/klib, src/msgpack). However,
     # we can lint all header files since they aren't "generated" like C files are.
     MSG='Linting .c and .h' ; echo $MSG
-    cpplint --quiet --extensions=c,h --headers=h --recursive --filter=-readability/casting,-runtime/int,-build/include_subdir pandas/_libs/src/*.h pandas/_libs/src/parser pandas/_libs/ujson pandas/_libs/tslibs/src/datetime
+    cpplint --quiet --extensions=c,h --headers=h --recursive --filter=-readability/casting,-runtime/int,-build/include_subdir pandas/_libs/src/*.h pandas/_libs/src/parser pandas/_libs/ujson pandas/_libs/tslibs/src/datetime /pandas/io/msgpack /pandas/_libs/*.cpp /pandas/util
```
[the remark this answers is not in the document]
Comment (saurav2608) [reply ci/code_checks.sh]: my bad.

**Comment by saurav2608** [comment]
@jreback can you please merge this?

**Comment by jreback** [comment]
i will look next day or 2

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @saurav2608

**Pull request merged**